Spigot-Server-Patches/0751-Fix-invulnerable-end-crystals.patch

```diff
@@ -0,0 +1,62 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Thu, 27 May 2021 14:52:30 -0700
+Subject: [PATCH] Fix invulnerable end crystals
+
+MC-108513
+
+diff --git a/src/main/java/net/minecraft/world/level/dimension/end/EnderDragonBattle.java b/src/main/java/net/minecraft/world/level/dimension/end/EnderDragonBattle.java
+index 0ab6319aa3e4e1f5679f37be36999ca56ca2484c..bf55d05681a719904ae5c272dad4290bf67e8979 100644
+--- a/src/main/java/net/minecraft/world/level/dimension/end/EnderDragonBattle.java
++++ b/src/main/java/net/minecraft/world/level/dimension/end/EnderDragonBattle.java
+@@ -570,6 +570,7 @@ public class EnderDragonBattle {
+ 
+         while (iterator.hasNext()) {
+             WorldGenEnder.Spike worldgenender_spike = (WorldGenEnder.Spike) iterator.next();
++            if (worldgenender_spike.crystal == null) { // Paper
+             List<EntityEnderCrystal> list = this.world.a(EntityEnderCrystal.class, worldgenender_spike.f());
+             Iterator iterator1 = list.iterator();
+ 
+@@ -579,6 +580,13 @@ public class EnderDragonBattle {
+                 entityendercrystal.setInvulnerable(false);
+                 entityendercrystal.setBeamTarget((BlockPosition) null);
+             }
++            // Paper start
++            } else {
++                worldgenender_spike.crystal.setInvulnerable(false);
++                worldgenender_spike.crystal.setBeamTarget(null);
++                worldgenender_spike.crystal = null;
++            }
++            // Paper end
```

Comment:
Why is this state even possible? I'm no fan of such a solution.

Reply:
I don’t know if this state is possible. I’m unclear on what happens if the chunks are unloaded during a dragon respawn, or any other fringe behavior so I thought best to leave the old logic in. 

Is your “no fan of such a solution” directed at the whole idea of the spikes keeping a reference to the entity, or at leaving the old reset logic?